### PR TITLE
Improve PyQt signals, mouse tracking, and more

### PR DESCRIPTION
**Pull request by banesullivan**
~this introduces an error in a downstream package that needs to be fixed~

I'm no PyQt expert, so someone needs to double-check my updates to the `QtInteractor` class (pinging @GuillaumeFavelier or @akaszynski)

New Features:

- mouse positions and/or click positions can be tracked automatically: `BasePlotter.track_mouse_position` and `BasePlotter.track_click_position`
  - Can easily stop tracking with `BasePlotter.untrack_*_position`
  - `BasePlotter.track_click_position` allows for a callback which is passed the viewport coordinates of the click position
- A picker was added to use the mouse/click position: `PickingHelper.pick_mouse_position`/`PickingHelper.pick_click_position` to return the 3D world coordinates of the mouse/click position
- A nice little helper method makes it easy to use the click event tracking and the picking features to fly to points in the scene with a right-click: `PickingHelper.enable_fly_to_right_click`
- Easy way to call the FXAA antialiasing method `BasePlotter.enable_anti_aliasing`
- A new keyboard shortcut `f` was added to fly to the mouse's current location in the scene
- A new style for rotating actors in the scene: `BasePlotter.enable_trackball_actor_style`,
- Some cleanup to the `pyqtSignal`s in the `QtInteractor` class

### CODE REVIEW

File: pyvista/plotting/qt_plotting.py [hunk 1]
```diff
@@ -323,6 +327,11 @@ def __init__(self, parent=None, title=None, shape=(1, 1), off_screen=None,
 
         self.signal_set_view_vector.connect(self.view_vector)
         self.signal_reset_camera.connect(self.reset_camera)
+        # self.render_trigger.connect(self._render)
+        self.signal_enable_trackball_style.connect(self.enable_trackball_style)
+        self.signal_remove_legend.connect(self.remove_legend)
+        self.signal_set_background.connect(self.set_background)
+        self.signal_remove_actor.connect(self.remove_actor)
```
Comment (GuillaumeFavelier) [inline pyvista/plotting/qt_plotting.py]: I see those signals are created and connected but when are they emitted?

Comment (banesullivan) [reply pyvista/plotting/qt_plotting.py]: Ooohhh. Okay I see now

I’m trying to teach myself proper PyQt development and these signals were confusing me. I’ll override those methods to emit a signal on the superclass

File: pyvista/plotting/plotting.py [hunk 2]
```diff
@@ -263,10 +337,17 @@ def update_style(self):
...
     def enable_trackball_style(self):
-        """ sets the interactive style to trackball - the default syle """
+        """ sets the interactive style to trackball camera - the default syle
+        """
         self._style = vtk.vtkInteractorStyleTrackballCamera()
         return self.update_style()
 
+    def enable_trackball_actor_style(self):
+        """ sets the interactive style to trackball actor. This makes it
+        possiblee to rotate actors around the scene."""
```
[the remark this answers is not in the document]
Comment (banesullivan) [reply pyvista/plotting/plotting.py]: Wow. I thought I had a general spell checker set up to lint in Atom... must not be working 🙃

File: pyvista/plotting/plotting.py [hunk 3]
```diff
@@ -215,6 +215,79 @@ def clear_events_for_key(self, key):
         self._key_press_event_callbacks.pop(key)
 
 
+    def enable_anti_aliasing(self, state=True):
```
[the remark this answers is not in the document]
Comment (banesullivan) [reply pyvista/plotting/plotting.py]: I definitely agree that `enable_*`/`disable_*` is more consistent. Will change

**Comment by banesullivan** [comment]
Thanks, @GuillaumeFavelier!

File: pyvista/plotting/qt_plotting.py (hunk 1 above)
Comment (banesullivan) [reply pyvista/plotting/qt_plotting.py]: Can anyone comment on when these signals are needed vs when they are not? Why do we need a signal for `set_background_color` and not for other things like setting the camera position?

Should we set up these signals for all methods?

**Pull request merged**